apps/demos/src/zone/zone-flags-configurator.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  ZoneGlobalDisableConfigurationsKey,
+  zoneGlobalDisableConfigurationsKeys,
+  ZoneRuntimeConfigurationsKey,
+  zoneRuntimeConfigurationsKeys,
+  ZoneTestDisableConfigurationsKey,
+  zoneTestDisableConfigurationsKeys,
+  ZoneTestSettingsConfigurationsKey,
+  zoneTestSettingsConfigurationsKeys
+} from './model/configurations.types';
+import { ZoneGlobalConfigurations } from './model/zone.configurations.api';
+
+type GlobalDisableConfigurationMethods = {
+  [disabledFlag in ZoneGlobalDisableConfigurationsKey]: (isDisabled?: boolean) => void;
+};
+
+type TestDisableConfigurationMethods = {
+  [disabledFlag in ZoneTestDisableConfigurationsKey]: (isDisabled?: boolean) => void
+} & {
+  [symbolFlag in ZoneTestSettingsConfigurationsKey]: (isDisabled?: boolean) => void
+};
+
+type targetSet = [WebSocket, (keyof WebSocketEventMap)[]] |
+  [any, (keyof WindowEventMap)[]];
+type GlobalTargetConfigurationMethod = (...args: targetSet) => void;
+type RuntimeConfigurationMethods = {
+  [disabledFlag in ZoneRuntimeConfigurationsKey]: (isDisabled?: boolean) => void;
+}
+
+const zoneDisable = '__Zone_disable_';
+const zoneIgnore = '__Zone_ignore';
+const zoneSymbol = '__zone_symbol__';
+
+
+export interface ZoneFlagsConfigurator {
```

Comment:
will there be a property named `enable` in future? otherwise I would reconsider the naming here

Reply:
No. disable could get false, but false is the default. As all this will be in one file I can't imagine setting it to false and then true again.

What would be a better naming?

Reply:
okay I got it wrong somehow. Improvements could only be made with some kind of `Record` and a ConditionalType which maps e.g. `global` to `GlobalDisableConfigurationMethods`. Lets go with this version for now i'd say. Otherwise we would need our typing magician @kajetansw